File: src/main/java/org/decembrist/mapper/UserToDtoMapper.java
package org.decembrist.mapper;

import org.decembrist.domain.User;
import org.decembrist.dto.UserDto;
import org.springframework.stereotype.Service;

//@Component
@Service
public class UserToDtoMapper implements Mapper<User, UserDto> {

	@Override
	public UserDto map(User source) {
		if (source == null) {
			return null;
		}
		final var userDto = new UserDto();
		userDto.setId(source.getId());
		userDto.setEmail(source.getEmail());
		userDto.setName(source.getUsername());
		return userDto;
	}

}

File: src/main/kotlin/org/decembrist/mapper/UserToDtoMapper.kt
package org.decembrist.mapper

import org.decembrist.domain.User
import org.decembrist.dto.UserDto
import org.springframework.stereotype.Service

//@Component
@Service
class UserToDtoMapper : Mapper<User?, UserDto?> {

    override fun map(source: User?): UserDto? {
        if (source == null) {
            return null
        }
        val userDto = UserDto()
        userDto.id = source.id
        userDto.email = source.email
        userDto.name = source.username
        return userDto
    }

}
